book/src/03_3_2_http_client.md

```diff
@@ -12,78 +10,63 @@ In this exercise, we'll write a small client that retrieves data over a HTTP con
+
+✅ Add the url `http://neverssl.com/` is in the main function. This is the address we will query.
 
-✅ Create a new `EspHttpClient` with default values.
+✅ Create a new `EspHttpClient` with default values (Try to find the method `new_default()` in the documentation).
 
-TODO currently no `docs.rs` documentation for the svc crates (ping Espressif about this, it's a known issue though)
 
-Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. This makes more sense with `POST` and `PUT` but must still be performed with `GET`. 
+Calling HTTP functions (e.g. `get(url)`) on this client returns an `EspHttpRequest`, which must be turned into a `Writer` to reflect the client's option to send some data alongside its request. 
 
 After this optional send step the `Writer` can be turned into a `Response` from which the received server output can be read:
 
-TODO explain as_ref(). Or find a better way to point people to it. 
 
 ```Rust
-let request = client.get(url.as_ref())?;
-// the parameter passed to `into_writer` is the number of bytes
-// the client intends to send
+let request = client.get(url)?;
 let writer = request.into_writer(0)?;
 let response = writer.into_response()?;
 ```
-✅ Implement sending a `GET` request and receiving a response.
+The parameter passed to `into_writer` is the number of bytes the client intends to send. Here we are not trying to send anything. 
 
 A successful response has [a status code in the 2xx range](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes).
 
 ✅ Verify the connection was successful.
 
 ✅ Return an Error if the status is not in the 2xx range.
 
+```rust
+match status {
+        200..=299 => {
+        }
+        _ => anyhow::bail!("unexpected response code: {}", status),
+    }
+```
+The status error can be returned with [Anyhow](https://docs.rs/anyhow/latest/anyhow/index.html), who returns the result and manages internally any type `Error` that implements `std::error:Error`.
+
 ✅ Turn your `response` into a `embedded_svc::io::Read` reader by calling `response.reader()` and read the received data chunk by chunk into a `u8` buffer using `reader.do_read(&mut buf)`. `do_read` returns the number of bytes read - you're done when this value is `0`.
 
 ✅ Report the total number of bytes read.
 
-✅ Log the received data to the console.
+✅ Log the received data to the console. Hint, the response in the buffert is in bytes, so you might need [a method](https://doc.rust-lang.org/std/str/fn.from_utf8.html) to convert from bytes to `&str`.
```

Comment:
explanation: "Hint" should not go into the instruction. If you think everyone needs this information, include it into the instruction. Otherwise a Hint section at the end of the page is better. 
"Might" sounds like you may or may not need it, this is a bit misleading as you really do need that. 
```suggestion
✅ Log the received data to the console. Use [from_utf8](https://doc.rust-lang.org/std/str/fn.from_utf8.html) to convert from bytes to `&str`.
```

Reply:
I'd add a note that this assumes valid UTF-8 from the server which technically isn't true - the encoding is sometimes sent in the HTTP headers, often isn't. (And then you have MIME types on top of that...) UTF-8 is a good default assumption though.